lib/services/remote_sync_service.dart

```diff
@@ -223,10 +224,87 @@ class RemoteSyncService {
   }
 
   Future<void> _syncDeviceCollectionFilesForUpload() async {
-    final devicePathCollections =
-        await FilesDB.instance.getDevicePathCollections();
+    final FilesDB fileDb = FilesDB.instance;
+    final devicePathCollections = await fileDb.getDevicePathCollections();
     devicePathCollections.removeWhere((element) => !element.sync);
     await _createCollectionsForDevicePath(devicePathCollections);
+    final Map<String, Set<String>> unSyncedPathIdToLocalIDs =
+        await fileDb.getDevicePathIDToLocalIDMap(syncStatus: false);
+    /*
+       A) Check if mapping for localID already exist in the collection
+       B) Check if
+     */
+    for (final eachDevicePath in devicePathCollections) {
```

Comment:
`for (final deviceCollection in deviceCollections)` will make it more readable as we go along